src/prefect/client/client.py

```diff
@@ -974,16 +980,41 @@ def update_task_run_heartbeat(self, task_run_id: str) -> None:
         }
         self.graphql(mutation, raise_on_error=True)
 
+    def get_flow_run_state(self, flow_run_id: str) -> "prefect.engine.state.State":
+        """
+        Retrieves the current state for a flow run.
+
+        Args:
+            - flow_run_id (str): the id for this flow run
+
+        Returns:
+            - State: a Prefect State object
+        """
+        query = {
+            "query": {
+                with_args("flow_run_by_pk", {"id": flow_run_id}): {
+                    "serialized_state": True,
+                }
+            }
+        }
+
+        flow_run = self.graphql(query).data.flow_run_by_pk
+
+        return prefect.engine.state.State.deserialize(flow_run.serialized_state)
+
     def set_flow_run_state(
-        self, flow_run_id: str, version: int, state: "prefect.engine.state.State"
+        self,
+        flow_run_id: str,
+        state: "prefect.engine.state.State",
+        version: int = None,
     ) -> "prefect.engine.state.State":
         """
         Sets new state for a flow run in the database.
 
         Args:
             - flow_run_id (str): the id of the flow run to set state for
-            - version (int): the current version of the flow run state
             - state (State): the new state for this flow run
+            - version (int, optional): the current version of the flow run state
```

Comment:
Could you note that the version is optional, but can be supplied to enforce version-locking